### PR TITLE
V0.8.1

### PR DESCRIPTION
**Pull request by Scarjit**
# Description

This release has sensorconnect and updates for our dependencies.

## Type of change

- [X] Release

### CODE REVIEW

**Comment by Scarjit** [comment]
# v0.8.1 sensorconnect on minikube
AFaUDNrOww

**Comment by Scarjit** [comment]
# Node-Red plugin
chrome_zZpkk2aVCn

**Comment by Scarjit** [comment]
@JeremyTheocharis v0.8.1 is ready for review

**Comment by Scarjit** [comment]
> SonarCloud Quality Gate failed.    Quality Gate failed
> 
> [Bug](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=SECURITY_HOTSPOT) [1 Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=CODE_SMELL) [7 Code Smells](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852) No Coverage information [](https://sonarcloud.io/component_measures?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&metric=new_duplicated_lines_density&view=list) [4.7% Duplication](https://sonarcloud.io/component_measures?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=852&metric=new_duplicated_lines_density&view=list)

This is irrelevant.
10.0.0.0/24 is a normal CIDR notation and does not expose any information

File: golang/cmd/sensorconnect/processIoddFiles.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+func contains(slice []string, entry string) bool {
+	for _, element := range slice {
+		if element == entry {
+			return true
+		}
+	}
+	return false
+}
+
+func getNamesOfFileInfo(fileInfoSlice []os.FileInfo) (namesSlice []string) {
```
Comment (JeremyTheocharis) [inline golang/cmd/sensorconnect/processIoddFiles.go:238]: function comment

File: golang/cmd/sensorconnect/processIoddFiles.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	for _, element := range fileInfoSlice {
+		namesSlice = append(namesSlice, element.Name())
+	}
+	return
+}
+
+func currentDateEarlierThenOldDate(currentDate string, oldDate string) (bool, error) {
```
Comment (JeremyTheocharis) [inline golang/cmd/sensorconnect/processIoddFiles.go:245]: function comment

File: golang/cmd/sensorconnect/processSensorData.go [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+	default:
+		outputString, _ := strconv.ParseUint(binaryValue, 2, 64)
+		return fmt.Sprintf("%v", outputString)
+	}
+	return
+}
+
+// createDigitalInputPayload creates a json output body from a DigitalInput to send via mqtt or kafka to the server
+func createDigitalInputPayload(serialNumber string, portNumberString string, timestampMs string, dataPin2In []byte) (payload []byte) {
+	payload = []byte(`{
```
Comment (JeremyTheocharis) [inline golang/cmd/sensorconnect/processSensorData.go:397]: serial_number is not required. you can remove it entirely

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]

File: golang/cmd/sensorconnect/downloadSensorData.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+		payload = append(payload, []byte(`]/iolinkdevice/deviceid",
+			"/iolinkmaster/port[`)...)
+		payload = append(payload, currentPort...)
+		payload = append(payload, []byte(`]/iolinkdevice/pdin",
+			"/iolinkmaster/port[`)...)
+		payload = append(payload, currentPort...)
+		payload = append(payload, []byte(`]/iolinkdevice/vendorid",
+			"/iolinkmaster/port[`)...)
+		payload = append(payload, currentPort...)
+		payload = append(payload, []byte(`]/pin2in"`)...)
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/downloadSensorData.go:86]: I have no clue
@larswitte

File: deployment/united-manufacturing-hub/values.yaml [hunk 5]
```diff
@@ -17,8 +16,16 @@ _000_commonConfig:
...
-      enabled: false
-      iprange: "127.0.0.1/32"  # e.g. 172.21.9.0/24
+      enabled: true
+      iprange: "192.168.10.1/24"  # IP Range in CIDR notation
+      ioddfilepath: "/ioddfiles"
+      useKafka: true
+      useMQTT: false
+      lowerTickSpeedMs: 100 # Fastest time, sensors should be requested to serve data
+      upperTickSpeedMs: 100 # Slowest time, sensors should be requested to serve data
+      TickSpeedSteppingUpMs: 10 # Time to add to actual stepping in case of failure
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply deployment/united-manufacturing-hub/values.yaml]: done

File: golang/cmd/sensorconnect/iolinkDeviceFinder.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	unmarshaledAnswer := DiscoverResponseFromDevice{}
+
+	// Unmarshal file with Unmarshal
+	err = json.Unmarshal(body, &unmarshaledAnswer)
+	if err != nil {
+		//zap.S().Errorf("Unmarshal of body from url %s failed.", url)
+		return
+	}
+	// Check CID
+	if unmarshaledAnswer.Cid != 23 {
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/iolinkDeviceFinder.go:77]: @larswitte

File: golang/cmd/sensorconnect/iolinkDeviceFinder.go [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+	ip := make(net.IP, 4)
+	binary.BigEndian.PutUint32(ip, i)
+
+	// converto ip to url
+	url = "http://" + ip.String()
+
+	// Payload to send to the gateways
+	var payload = []byte(`{
+        "code":"request",
+        "cid":23,
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/iolinkDeviceFinder.go:137]: @larswitte

File: golang/cmd/sensorconnect/main.go [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+	actualSensorTickSpeed = lowestSensorTickTime
+	zap.S().Infof("Requesting data every %d ms", actualSensorTickSpeed)
+	tickerProcessSensorData := time.NewTicker(time.Duration(actualSensorTickSpeed) * time.Millisecond)
+	defer tickerProcessSensorData.Stop()
+	go continuousSensorDataProcessing(tickerProcessSensorData, updateIoddIoDeviceMapChan)
+
+	select {} // block forever
+}
+
+func continuousSensorDataProcessing(ticker *time.Ticker, updateIoddIoDeviceMapChan chan IoddFilemapKey) {
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/main.go]: Added comment in src

File: golang/cmd/sensorconnect/main.go [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+			}
+
+			if hadError && actualSensorTickSpeed == upperSensorTickTime {
+				zap.S().Warnf("One of your sensors is to slow or died !")
+			}
+		}
+	}
+}
+
+func downloadSensorDataMapAndProcess(deviceInfo DiscoveredDeviceInformation, updateIoddIoDeviceMapChan chan IoddFilemapKey, portModeMap map[int]int) {
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/main.go]: Added comment in src

File: golang/cmd/sensorconnect/main.go [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+	var sensorDataMap map[string]interface{}
+	var err error
+	sensorDataMap, err = GetSensorDataMap(deviceInfo)
+	if err != nil {
+		return
+	}
+	processSensorData(deviceInfo, updateIoddIoDeviceMapChan, portModeMap, sensorDataMap)
+}
+
+func continuousDeviceSearch(ticker *time.Ticker, ipRange string) {
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/main.go:302]: Added comment in src

File: golang/cmd/sensorconnect/main.go [hunk 11]
```diff
@@ -0,0 +1,289 @@
...
+			err = DiscoverDevices(ipRange)
+			if err != nil {
+				zap.S().Errorf("DiscoverDevices produced the error: %v", err)
+				continue
+			}
+		}
+	}
+
+}
+func ioddDataDaemon(updateIoddIoDeviceMapChan chan IoddFilemapKey, relativeDirectoryPath string) {
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/main.go:321]: Added comment in src

File: golang/cmd/sensorconnect/main.go [hunk 12]
```diff
@@ -0,0 +1,289 @@
...
+	}
+	fileInfo, err = AddNewDeviceToIoddFilesAndMap(siemensIoddFilemapKey, relativeDirectoryPath, fileInfo)
+	if err != nil {
+		zap.S().Errorf("AddNewDeviceToIoddFilesAndMap produced the error: %v", err)
+	}
+
+	return
+}
+
+func GetSyncMapLen(p *sync.Map) int {
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/main.go:372]: Added comment in src

File: golang/cmd/sensorconnect/mqtt.go [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+func SetupMQTT(certificateName string, mqttBrokerURL string, podName string) {
+	if !useMQTT {
+		return
+	}
+
+	opts := MQTT.NewClientOptions()
+	opts.AddBroker(mqttBrokerURL)
+	if certificateName == "NO_CERT" {
+		opts.SetClientID(podName)
+		opts.SetUsername("MQTT_KAFKA_BRIDGE")
```
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/mqtt.go]: Fixed, changed to SENSORCONNECT

File: golang/cmd/sensorconnect/processIoddFiles.go (hunk 1 above)
Comment (Scarjit) [reply golang/cmd/sensorconnect/processIoddFiles.go:235]: Added comment in src

File: golang/cmd/sensorconnect/processIoddFiles.go (hunk 2 above)
Comment (Scarjit) [reply golang/cmd/sensorconnect/processIoddFiles.go:243]: Added comment in src

File: golang/cmd/sensorconnect/processSensorData.go (hunk 3 above)
Comment (Scarjit) [reply golang/cmd/sensorconnect/processSensorData.go]: fixed

File: golang/cmd/sensorconnect/iolinkDeviceFinder.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/iolinkDeviceFinder.go]: Added comment that CID can be any number

File: golang/cmd/sensorconnect/iolinkDeviceFinder.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (Scarjit) [reply golang/cmd/sensorconnect/iolinkDeviceFinder.go:133]: Adde comment that CID can be any number

**Review by JeremyTheocharis — APPROVED** [review]

**Pull request merged**